notebooks/create_htmls.ipynb

```diff
@@ -175,126 +457,274 @@
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    edu_exp_dir : str or pathlib.Path\n",
+    "        The directory path containing \"experiences.json\" and \"education.json\" files.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pandas.DataFrame\n",
+    "        A DataFrame containing filtered and preprocessed education and experience data.\n",
+    "    \"\"\"\n",
+    "    edu_exp_list = []\n",
     "    file_names = [\"experiences.json\", \"education.json\"]\n",
     "    valid_groups = [\"DTI\", \"TARDIS\", \"ICER\", \"kerzendorf\"]\n",
     "    valid_institution = \"Michigan State University\"\n",
-    "\n",
     "    for file_name in file_names:\n",
-    "        file_path = member_json_dir / file_name\n",
+    "        file_path = edu_exp_dir / file_name\n",
     "        if file_path.exists():\n",
     "            # Reading JSON data directly into a DataFrame\n",
     "            df = pd.read_json(file_path)\n",
     "            \n",
     "            # filtering based on group and institution\n",
-    "            mask = df.apply(lambda x: x.get('group') in valid_groups or x.get('institution') == valid_institution, axis=1)\n",
-    "            filtered_df = df[mask]\n",
+    "            filtered_df = filter_edu_exp_data(df, valid_groups, valid_institution)\n",
     "            \n",
-    "            education_experience_list.append(filtered_df)\n",
+    "            edu_exp_list.append(filtered_df)\n",
     "        else:\n",
     "            print(f\"{file_path} does not exist\")\n",
     "\n",
-    "    if education_experience_list:\n",
-    "        education_experience_df = pd.concat(education_experience_list, ignore_index=True)\n",
+    "    if edu_exp_list:\n",
+    "        edu_exp_df = pd.concat(edu_exp_list, ignore_index=True)\n",
     "    else:\n",
-    "        education_experience_df = pd.DataFrame()\n",
+    "        edu_exp_df = pd.DataFrame()\n",
     "\n",
     "    # if start_date column exists, fill with NaN if it doesn't\n",
-    "    if 'start_date' not in education_experience_df:\n",
-    "        education_experience_df['start_date'] = pd.NaT\n",
+    "    if 'start_date' not in edu_exp_df:\n",
+    "        edu_exp_df['start_date'] = pd.NaT\n",
     "    \n",
     "    # Convert start_date to datetime format\n",
-    "    education_experience_df['start_date'] = pd.to_datetime(education_experience_df['start_date'], errors='coerce')\n",
+    "    edu_exp_df['start_date'] = pd.to_datetime(edu_exp_df['start_date'], errors='coerce')\n",
     "    \n",
     "    # Sort the DataFrame based on start_date\n",
-    "    education_experience_df = education_experience_df.sort_values(by='start_date', ascending=False)\n",
-    "\n",
-    "    # Load social links directly\n",
-    "    social_links = {}\n",
-    "    social_links_file_path = member_json_dir / \"social_links.json\"\n",
+    "    edu_exp_df = edu_exp_df.sort_values(by='start_date', ascending=False)\n",
```

Comment:
I think you are sorting the dataframe twice. Can you check?

Reply:
We are not sorting the data frame twice. Currently, the data is already presented in a sorted manner in the JSON files. If someone, by mistake enters the data in the wrong order then the "sort_values()" will automatically handle it.

Reply:
I see.